receiver/mongodbreceiver/extractor.go

```diff
@@ -0,0 +1,247 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver/internal/metadata"
+)
+
+type extractor struct {
+	version *version.Version
+	logger  *zap.Logger
+}
+
+type extractType int
+
+const (
+	adminServerStats extractType = iota
+	normalDBStats
+	normalServerStats
+)
+
+// Mongo26 is a version representation of mongo 2.6
+var Mongo26, _ = version.NewVersion("2.6")
+
+// Mongo30 is a version representation of mongo 3.0
+var Mongo30, _ = version.NewVersion("3.0")
+
+// Mongo40 is a version representation of mongo 4.0
+var Mongo40, _ = version.NewVersion("4.0")
+
+// Mongo42 is a version representation of mongo 4.2
+var Mongo42, _ = version.NewVersion("4.2")
+
+// Mongo44 is a version representation of mongo 4.4
+var Mongo44, _ = version.NewVersion("4.4")
+
+// Mongo50 is a version representation of mongo 5.0
+var Mongo50, _ = version.NewVersion("5.0")
```

Comment:
I don't think any of these are necessary. Most aren't used at all. The others are used very few times. Just use `version.NewVersion("4.4")` when necessary.

```suggestion
// Mongo26 is a version representation of mongo 2.6
var Mongo26, _ = version.NewVersion("2.6")

// Mongo30 is a version representation of mongo 3.0
var Mongo30, _ = version.NewVersion("3.0")

// Mongo40 is a version representation of mongo 4.0
var Mongo40, _ = version.NewVersion("4.0")

// Mongo42 is a version representation of mongo 4.2
var Mongo42, _ = version.NewVersion("4.2")

// Mongo44 is a version representation of mongo 4.4
var Mongo44, _ = version.NewVersion("4.4")

// Mongo50 is a version representation of mongo 5.0
var Mongo50, _ = version.NewVersion("5.0")
```